src/main/java/com/google/cloud/tools/ide/login/GoogleLoginState.java

```diff
@@ -253,22 +254,24 @@ public boolean logOut(boolean showPrompt) {
     return true;
   }
 
-  private Credential makeCredential(String accessToken, String refreshToken) {
-    Credential cred =
-        new GoogleCredential.Builder()
-            .setJsonFactory(jsonFactory)
-            .setTransport(transport)
-            .setClientSecrets(clientId, clientSecret)
-            .build();
-    cred.setAccessToken(accessToken);
-    cred.setRefreshToken(refreshToken);
-    return cred;
+  private Credential makeCredential(
+      String accessToken, String refreshToken, long expiresInSeconds) {
```

Comment:
I'd change `expiresInSeconds` to `expiryInMilliseconds` relative to `System.currentTimeMillis()`.  `Credential` has  `[gs]etExpirationTimeInMilliseconds()` (absolute, relative to `System.currentTimeMillis()`), and `[gs]etExpiresInSeconds()` (relative to now).  The differences between their names is very subtle!

I think using the expiry-in-milliseconds rather than expires-in would simplify your code quite a bit.